src/test/java/com/shapesecurity/salvation/PolicyQueryingTest.java

```diff
@@ -1316,5 +1320,48 @@
         assertFalse(p.allowsFrameFromSource(URI.parse("http://example.com")));
         assertTrue(p.allowsWorkerFromSource(URI.parse("http://example.com")));
         assertTrue(p.allowsScriptFromSource(URI.parse("http://example.com")));
+
+        p = Parser.parse(" worker-src blob:", "http://example.com");
+        assertTrue(p.allowsWorkerFromSource(new GUID("blob:")));
+        assertFalse(p.allowsWorkerFromSource(new GUID("data:")));
+    }
+
+    @Test public void testAllowNavigationTo() {
+        Policy p = Parser.parse("navigate-to blob:", "http://example.com");
+        assertTrue(p.allowNavigationTo(new GUID("blob:")));
+        assertFalse(p.allowNavigationTo(new GUID("data:")));
+        assertTrue(p.allowFormActionTo(new GUID("blob:")));
+        assertFalse(p.allowFormActionTo(new GUID("data:")));
+
+        p = Parser.parse("navigate-to blob:; form-action data:", "http://example.com");
+        assertTrue(p.allowNavigationTo(new GUID("blob:")));
+        assertFalse(p.allowNavigationTo(new GUID("data:")));
+        assertTrue(p.allowFormActionTo(new GUID("data:")));
+        assertFalse(p.allowFormActionTo(new GUID("blob:")));
+
+        p = Parser.parse("form-action data:", "http://example.com");
+        assertFalse(p.allowNavigationTo(new GUID("blob:")));
+        assertFalse(p.allowNavigationTo(new GUID("data:")));
+        assertTrue(p.allowFormActionTo(new GUID("data:")));
+        assertFalse(p.allowFormActionTo(new GUID("blob:")));
+
+
+        p = Parser.parse("navigate-to a", "http://example.com");
+        assertTrue(p.allowNavigationTo(URI.parse("http://a")));
+        assertFalse(p.allowNavigationTo(URI.parse("http://b")));
+        assertTrue(p.allowFormActionTo(URI.parse("http://a")));
+        assertFalse(p.allowFormActionTo(URI.parse("http://b")));
+
+        p = Parser.parse("navigate-to a; form-action b", "http://example.com");
+        assertTrue(p.allowNavigationTo(URI.parse("http://a")));
+        assertFalse(p.allowNavigationTo(URI.parse("http://b")));
+        assertTrue(p.allowFormActionTo(URI.parse("http://b")));
+        assertFalse(p.allowFormActionTo(URI.parse("http://a")));
+
+        p = Parser.parse("form-action a", "http://example.com");
+        assertFalse(p.allowNavigationTo(URI.parse("http://a")));
```

Comment:
These should be `true` since `navigate-to` is missing and `navigate-to` does not defer to any other directive.